docs/jumpstart/week-9.md

```diff
@@ -0,0 +1,343 @@
+
+But storing our data is only one part of our application. Equally important is accessing it. Let's say we want to list all the products in an order. The first step is to obtain all the
+ `product_id`s associated with that order. So, we might run:
+ 
+```sql
+SELECT * FROM order_items WHERE order_id = 20
+```
+
+to find all the order_items associated with order 20. Then, for each order item, we can find its product by running separate queries:
+ 
+```sql
+SELECT * FROM products WHERE id = ORDER_ITEM_ID
+```
+
+but this verbose. Instead, we can use the `JOIN` keyword, which allows us to join two tables
+ based on a matching column, like an id. So, we could write:
+ 
+```sql
+SELECT products.id AS Product_ID, products.name AS Product_Name, orders.id AS Order_ID
+FROM order_items 
+    JOIN products
+ON 
+    Product_ID = order_items.product_id;
+```
+
+This table will give us 3 colums representing the Product ID in our order, the name of the product, and finally our order ID in case we wanted to double check that we've got the right products. The rows will contain the list of products in our order.
+
+### Different types of joins
+
+<img align="right" src="img/Joins-in-SQL-Inner-Outer-Left-and-Right-Join.jpg" width="300px">
+
+Based on what we want to get from two tables, we can do different types of joins. As you can see in the picture below, some of these different joins are: `INNER JOIN` (or `JOIN` as default in SQL), `LEFT JOIN`, `RIGHT JOIN`, and `FULL JOIN`. 
+
+`INNER JOIN`: returns rows when there is a match in both tables.
+
+`LEFT JOIN`: returns all rows from the left table, even if there are no matches in the right table.
+
+`RIGHT JOIN`: returns all rows from the right table, even if there are no matches in the left table.
+
+`FULL JOIN`: It combines the results of both left and right outer joins.
+The joined table will contain all records from both the tables and fill in NULLs for missing matches on either side.
+
+
+## Subquery
+Subquery's definition is within its name: a query within a query! Subqueries could be nested inside a `SELECT`, `INSERT`, `UPDATE`, `DELETE`, or another subquery. They can output a list of data or individual values. 
+
+We use subqueries as a condition (therefore it's usually used within a `WHERE` clause) and will return data that will be used in the main query and the point of it is to restrict the information that's being retrieved from the database. Make sure you surround your subqueries with paranthesis.
+
+Note: Operators could be < = > etc.
+```sql
+SELECT column1, column2, ...
+FROM table
+WHERE clause/expressions OPERATOR 
+    (SELECT column
+    FROM table
+    WHERE condition);
+```
```

Comment:
Can you add an example too?

Reply:
Also, can you mention that you can add a table alias like in joins so that you can reference the result in other parts of the query (like in select)